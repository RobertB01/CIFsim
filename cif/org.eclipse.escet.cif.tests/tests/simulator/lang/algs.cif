//////////////////////////////////////////////////////////////////////////////
// Copyright (c) 2010, 2020 Contributors to the Eclipse Foundation
//
// See the NOTICE file(s) distributed with this work for additional
// information regarding copyright ownership.
//
// This program and the accompanying materials are made available
// under the terms of the MIT License which is available at
// https://opensource.org/licenses/MIT
//
// SPDX-License-Identifier: MIT
//////////////////////////////////////////////////////////////////////////////

alg int a0 = a1;                         // 1

alg int a1 = 1;

alg int a2;
equation a2 = p.v1 + a1;                 // p.v1 + 1

automaton p:
  disc int v1 = a0 + 1;                  // 2

  location:
    initial;
    urgent;

    edge when a2 > 0 do v1 := v1 - 1;    // 3 times, v1 becomes: 1, 0, -1
end
