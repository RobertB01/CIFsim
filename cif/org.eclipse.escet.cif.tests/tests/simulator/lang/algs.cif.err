WARNING: File "simulator/lang/algs.cif": Semantic warning at line 14, column 14: Type "int[1..1]" allows only one value.
WARNING: File "simulator/lang/algs.cif": Semantic warning at line 16, column 14: Type "int[1..1]" allows only one value.
