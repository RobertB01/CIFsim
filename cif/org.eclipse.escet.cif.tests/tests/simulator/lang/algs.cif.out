Initial state: time=0.0, a0=1, a1=1, a2=3, p=*, p.v1=2

Transition: event tau
State: time=0.0, a0=1, a1=1, a2=2, p=*, p.v1=1

Transition: event tau
State: time=0.0, a0=1, a1=1, a2=1, p=*, p.v1=0

Transition: event tau
State: time=0.0, a0=1, a1=1, a2=0, p=*, p.v1=-1

Simulation resulted in deadlock.
